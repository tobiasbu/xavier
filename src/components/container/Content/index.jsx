import React from 'react';
import PropTypes from 'prop-types';

import useStyle from './style';

/**
 * Page content.
 */
const Content = (props) => {
  const { children } = props;

  const classes = useStyle();

  return (
    <div className={classes.contentContainer}>
      {children}
    </div>
  );
};

/**
 * Content prop types
 */
Content.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Content;
