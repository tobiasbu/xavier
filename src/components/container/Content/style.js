import { createUseStyles } from 'react-jss';
import MediaQueries from '../../../style/mediaQueries';

const useStyle = createUseStyles({
  contentContainer: {
    position: 'relative',
    flex: '1 1 0%',
    margin: '0 4rem',
    maxWidth: '48rem',
    [MediaQueries.large]: {
      maxWidth: 'rem',
    },
  },
});

export default useStyle;
