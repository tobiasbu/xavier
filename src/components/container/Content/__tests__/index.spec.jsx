import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Content from '..';

describe('Content', () => {
  it('should render', () => {
    const component = shallow(
      <Content>
        <div>
          Feij√£o
          <span>Arroz</span>
          Mandioca
        </div>
      </Content>,
    );
    expect(toJSON(component)).toMatchSnapshot();
  });
});
