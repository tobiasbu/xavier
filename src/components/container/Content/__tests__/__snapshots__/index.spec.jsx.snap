// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render 1`] = `
<div
  className="contentContainer-0-2-1"
>
  <div>
    Feij√£o
    <span>
      Arroz
    </span>
    Mandioca
  </div>
</div>
`;
